### PR TITLE
more functions

### PR DESCRIPTION
**Pull request by Cofflag**

### CODE REVIEW

File: include/attribute.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#ifndef GUARD_ATTRIBUTE_H
+#define GUARD_ATTRIBUTE_H
+
+#include "global.h"
+
+struct Roleattrib
```
Comment (jiangzhengwenjz) [inline include/attribute.h:6]:
```suggestion
struct RoleAttrib
```

File: src/attribute.c [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#include "global.h"
+#include "attribute.h"
+#include "gba/syscall.h"
```
Comment (jiangzhengwenjz) [inline src/attribute.c:3]: I believe you don't need to include it when having global.h

File: include/attribute.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+{
+    u8 filler0[0x78];
+};
+
+extern u32 *gUnk_03006994; //TODO:Correct type.
+extern u32 gUnk_02001000;
```
Comment (jiangzhengwenjz) [inline include/attribute.h:12]: This is almost surely incorrect. 
```
        . = 0x00001000; gUnk_02001000 = .;
        . = 0x00001800; gUnk_02001800 = .;
```

File: include/attribute.h [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+extern const struct Roleattrib *gpAttribTable;
```
Comment (jiangzhengwenjz) [inline include/attribute.h:13]: This is not gPascalCase. You probably want something like `gAttribTablePtr`

File: include/attribute.h [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+extern const struct Roleattrib gAttribTable[];
+
+/* 
+Why can I point a struct pointer to a struct array type? 
+I get this from the internet, but I really don't understand. 
+Isn't it unable to point something that has a different type? 
+Don't understand. 
+*/
```
Comment (jiangzhengwenjz) [inline include/attribute.h:21]: A comment is for explaining code. If you have fundamental question, you can directly ask me.

File: include/attribute.h [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+extern void sub_080154F0();
```
Comment (jiangzhengwenjz) [inline include/attribute.h:23]:
```suggestion
extern void sub_080154F0(void);
```

**Review by jiangzhengwenjz — CHANGES_REQUESTED** [review]

**Comment by Cofflag** [comment]
The struct is big, so I use printf("0x%x",sizeof(struct SaveBlock1);to check and fix the size.

File: include/attribute.h [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+    u8 boxGloveForgeLevel;
+    u8 bowForgeLevel;
+    u16 filler8[0xE]; //Something related to player name
+    u16 playerMaxHp; //max = 0x270f,but more is ok
+    u16 playerCurrentHp;
+    u16 equippedOneAtk; //max = 0x3e7
+    u16 equippedTwoAtk;
+    u16 equippedThreeAtk;
+    u16 notEquippedAtk;
+    u16 filler30; //TODO:maybe this is a substruct align
```
Comment (jiangzhengwenjz) [inline include/attribute.h:49]: In English, you need a space after `:`. Please fix other cases as well.

File: include/attribute.h [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+    u32 guardimalNextExp;
+    u8 battleItemOne;
+    u8 battleItemTwo;
+    u8 battleItemThree;
+    u8 battleItemFour;
+    u8 filler90[0x54];
+    struct BagWeapon bagWeapon[0x1E]; 
+    u8 filler42C[0x1E];
+    u8 numWoundBand;
+    u8 numJvhuaCha; // TODO:rename this
```
Comment (jiangzhengwenjz) [inline include/attribute.h:91]: I'm really confused because sometimes you have a space after `//` but sometimes not.

File: include/attribute.h [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+    u8 entrustFail2;
+    u8 entrustFail3;
+    u8 shifusWeapon;
+    u8 mightBook1;
+    u8 mightBook2;
+    u8 mightBook3;
+    u8 mightBook4;
+    u8 mightBook5;
+    u8 craftMightBook;
+    u8 mightBook6; // looks wierd.maybe need a fix
```
Comment (jiangzhengwenjz) [inline include/attribute.h:241]: Same thing. Need space after `.`

File: include/attribute.h [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+    u8 weaponDex[0x1E]; // This is a bit field. 1 = dex lighted, 0 = not lighted.
+    u8 filler582; // wierd
+    u8 weaponSkillDex1[6];
+    u8 weaponSkillDex2[6]; //maybe a useless checksum, need more evidence
+    u8 weaponSpecialEffectDex1[5];
+    u8 filler594;
+    u8 weaponSpecialEffectDex2[5];
+    u8 filler59A;
+    u8 summonGuardianDex[17];
+    u32 money; // max = 0x0098967F
```
Comment (jiangzhengwenjz) [inline include/attribute.h:267]: not readable. Why not just write `9999999`?

**Review by jiangzhengwenjz — CHANGES_REQUESTED** [review]

File: include/attribute.h (hunk 10 above)
Comment (Cofflag) [reply include/attribute.h:267]: I think there must be a sort of function like a CheckMoney(), so listing the hex value here may be convenient ^_^

**Review by jiangzhengwenjz — APPROVED** [review]

**Pull request merged**